pkg/controller/master/template/template_version_controller.go

```diff
@@ -70,39 +69,32 @@ func (h *templateVersionHandler) OnChanged(key string, tv *apisv1alpha1.VirtualM
 
 	//set version
 	if !apisv1alpha1.VersionAssigned.IsTrue(copyObj) {
-		existLatestVersion, err := h.getTemplateLatestVersion(tv.Namespace, tv.Name, tv.Spec.TemplateID)
+		selector := fmt.Sprintf("metadata.name!=%s", tv.Name)
+		existLatestVersion, _, err := getTemplateLatestVersion(tv.Namespace, tv.Spec.TemplateID, selector, h.templateVersions)
```

Comment:
It looks obscure to pass such a selector string as the function parameter. Why do we want to extract it out?